bundles/org.openhab.ui/web/src/assets/definitions/blockly/utils.js

```diff
@@ -201,3 +201,42 @@ export function addDateComparisonSupportGraalVM () {
     ])
   return graalZdtCompare
 }
+
+// always return the item name no matter whether the name is given already or the item object
+export function addItemName (isGraalJs) {
+  let itemNameAccess = (isGraalJs) ? 'name' : 'getName()'
```

Comment:
For GraalJS, I'd prefer that this method gets never used. Instead of using this method, please let me know where you need to be able to pass in both the Item and the Item name and I'll extend the openhab-js library to make that possible.

Reply:
Yes, I appreciate that: go ahead and do so. I will then tell you for the upcoming PR for all places where this would be applicable ahead of time, so we can jointly optimize that.